tools/floodgate/js/floodgate.js

```diff
@@ -39,9 +39,23 @@ async function refreshPage(config, projectDetail, project) {
 }
 
 function setListeners(project, projectDetail) {
+  const modal = document.getElementById('fg-modal');
   document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
-  document.querySelector('#copyFiles button').addEventListener('click', () => floodgateContent(project, projectDetail));
-  document.querySelector('#promoteFiles button').addEventListener('click', () => promoteFloodgatedFiles(project));
+  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => {
+      modal.style.display = 'none'; floodgateContent(project, projectDetail);
+    });
+    modal.style.display = 'block';
+  });
+  document.querySelector('#promoteFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => {
+      modal.style.display = 'none'; promoteFloodgatedFiles(project);
+    });
+    modal.style.display = 'block';
+  });
```

Comment:
I see what you're saying. I hadn't noticed that it was the same modal button. In that case, I guess it's fine to add the listeners within the other like you're doing, but you really should be removing them. It's best practice to remove listeners when you don't need them anyway. The code below is what I'd do.

Also, you'll notice that the variables should go at the top of the block, the event listeners should go at the bottom of their blocks, and anything with a semicolon should be on it's own line

```
  const handleFloodgateConfirm = ({ target }) => {
    modal.style.display = 'none';
    floodgateContent(project, projectDetail);
    target.removeEventListener('click', handleFloodgateConfirm);
  };
  const handlePromoteConfirm = ({ target }) => {
    modal.style.display = 'none';
    promoteFloodgatedFiles(project);
    target.removeEventListener('click', handlePromoteConfirm);
  };
  ...
  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
    modal.style.display = 'block';
    document.querySelector('#fg-modal #yes-btn').addEventListener('click', handleFloodgateConfirm);
  });
  document.querySelector('#promoteFiles button').addEventListener('click', (e) => {
    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
    modal.style.display = 'block';
    document.querySelector('#fg-modal #yes-btn').addEventListener('click', handlePromoteConfirm);
  });
```

Reply:
If you're curious, a lot of these points come from the airbnb style guide, which our linter rules are based on https://github.com/airbnb/javascript

Reply:
Thanks @meganthecoder. Added these changes now.